src/Approximations/2D_approximations.jl

```diff
@@ -0,0 +1,79 @@
+"""
+    overapproximate(S::LazySet, ::Type{HPolygon})::HPolygon
+
+Return an approximation of a given 2D convex set as a box-shaped polygon.
+
+### Input
+
+- `S` -- convex set, assumed to be two-dimensional
+- `HPolygon` for dispatch
+
+### Output
+
+A box-shaped polygon in constraint representation.
+"""
+function overapproximate(S::LazySet, ::Type{HPolygon})::HPolygon
```

Comment:
i don't understand, where is the second argument used?

Reply:
nevermind. i think i what i wanted to suggest is defining

```julia
overapproximate(S::LazySet) = overapproximate(S::LazySet, ::Type{HPolygon})
```

Reply:
> where is the second argument used?

It's only used by Julia to find the correct method.

Reply:
> maybe define
> `overapproximate(S::LazySet) = overapproximate(S, HPolygon)`

Good idea, I will try.